flow/project.py

```diff
@@ -881,25 +1051,30 @@ def _generate_id(self, job, operation_name=None, index=0):
         if max_len < len(job_op_id):
             raise ValueError("Value for MAX_LEN_ID is too small ({}).".format(self.MAX_LEN_ID))
 
+        if len(jobs) > 1:
+            aggregate_id = str(jobs[0])[:8]+'-'+str(jobs[-1])[:8]
+        else:
+            aggregate_id = str(jobs[0])[:8]
```

Comment:
This id does not seem sufficiently unique to me.

Reply:
I don't think that it's necessary for this particular id to be unique.
This is used for the "readable" part of the ID. We will store every aggregate associated with jobs in #335 hence if users want, they can look up aggregates associated with that particular id in the `.aggregates/` directory

Reply:
I was not aware of #335, however in that case this variable name is misleading. An id is by definition describing the *identity*, so it **must** be unique. If this string is describing something else, then that should be reflected in the semantics here. However, I see no reason why we are not showing the actual id here?

Reply:
I can remove the substring part of the job in jobs(`[:8]`) and instead use the full job id, but if we decide to include all the job ids then this will get way too long for large aggregates.

Reply:
Please reread my comment. If it is not necessary for this to be an id, that's fine, but then please do not call id *id*.